src/AuditReader.php

```diff
@@ -1022,7 +1022,7 @@ private function createEntity($className, array $columnMap, array $data, $revisi
                     foreach (self::getRelationToSourceKeyColumns($assoc) as $sourceKeyJoinColumn => $sourceKeyColumn) {
                         $whereId[] = "{$sourceKeyJoinColumn} = ?";
 
-                        $reflField = $classMetadata->reflFields['id'];
+                        $reflField = $classMetadata->reflFields[$classMetadata->getFieldName($sourceKeyColumn) ?? 'id'];
```

Comment:
If I'm not wrong, I guess we could use something like this:
```suggestion
                        $reflField = $classMetadata->getSingleIdReflectionProperty();
```

Reply:
Yes, you are right, this is much better approach and it seems to work in the same way.